### PR TITLE
add testGetNewToken() test

### PR DESCRIPTION
**Pull request by KarishmaSinghChauhan**

### CODE REVIEW

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 1]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
     }
+
+    @Test
+    public void testGetNewToken() {
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+
+        UUID shopId = inMemoryVirtualQueueRepository.createShop(shop);
+        Token newToken = inMemoryVirtualQueueRepository.getNewToken(shopId);
+        UUID tokenId = newToken.getTokenId();
+        int tokenMapSize = inMemoryVirtualQueueRepository.getTokenMap().size();
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:83]: rename to expectedTokenMapSize, expectedTokenNumber, etc

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 2]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
     @Test
     public void testGetWaitingTime() throws Exception {
         long waitingTime = inMemoryVirtualQueueRepository.getWaitingTime();
-        
+
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetNewToken() {
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:77]: rename to testGetNewToken_success

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 3]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
+
     @Test
     public void testGetWaitingTime() throws Exception {
         long waitingTime = inMemoryVirtualQueueRepository.getWaitingTime();
-        
+
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:76]: Follow the AAA pattern
https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 2 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 1 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Changed names.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 3 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:76]: Changed.

File: frontendservice/package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "bulma": "^0.9.0",
     "core-js": "^3.6.5",
     "firebase": "^7.15.0",
+    "js-cookie": "^2.2.1",
```
Comment (sdumka) [inline frontendservice/package.json:15]: Why is this needed?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 5]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
+    public void testGetNewToken_success() {
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        UUID shopId = inMemoryVirtualQueueRepository.createShop(shop);
+        Token newToken = inMemoryVirtualQueueRepository.getNewToken(shopId);
+        
+        UUID tokenId = newToken.getTokenId();
+        int actualTokenMapSize = inMemoryVirtualQueueRepository.getTokenMap().size();
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:83]: replace actual with expected. Here and in other variable names

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 6]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
-        
+
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetNewToken_success() {
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        UUID shopId = inMemoryVirtualQueueRepository.createShop(shop);
+        Token newToken = inMemoryVirtualQueueRepository.getNewToken(shopId);
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:80]: This line is the act block right? Can we add an empty line above?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 7]
```diff
@@ -63,11 +65,27 @@ public void testCreateShop() throws Exception {
...
+        
+        UUID tokenId = newToken.getTokenId();
+        int actualTokenMapSize = inMemoryVirtualQueueRepository.getTokenMap().size();
+        Integer actualTokenNumber = inMemoryVirtualQueueRepository.getTokenMap().get(tokenId).getTokenNumber();
+        UUID actualShopId = inMemoryVirtualQueueRepository.getTokenMap().get(tokenId).getShopId();
+
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:86]: Since there's an empty line in the Act block, add "// Arrange" "// Act" "// Assert" comments before starting each block

File: frontendservice/package.json (hunk 4 above)
Comment (KarishmaSinghChauhan) [reply frontendservice/package.json:15]: I had just pulled the entire code from master and this change came automatically. I haven't written this or used it.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 7 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 6 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:82]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 5 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Done.

File: frontendservice/package.json (hunk 4 above)
Comment (sdumka) [reply frontendservice/package.json:15]: remove this then?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 8]
```diff
@@ -63,11 +65,28 @@ public void testCreateShop() throws Exception {
...
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetNewToken_success() {
+        // Arrange
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        UUID shopId = inMemoryVirtualQueueRepository.createShop(shop);
+        // Act
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:81]: Empty line above Act and Assert comment lines

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 9]
```diff
@@ -63,11 +65,28 @@ public void testCreateShop() throws Exception {
...
+        Token newToken = inMemoryVirtualQueueRepository.getNewToken(shopId);
+        UUID tokenId = newToken.getTokenId();
+        int expectedTokenMapSize = inMemoryVirtualQueueRepository.getTokenMap().size();
+        Integer expectedTokenNumber = inMemoryVirtualQueueRepository.getTokenMap().get(tokenId).getTokenNumber();
+        UUID expectedShopId = inMemoryVirtualQueueRepository.getTokenMap().get(tokenId).getShopId();
+        // Assert
+        assertEquals("Size of tokenMap", 1, expectedTokenMapSize);
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:88]: Static variable "1"?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 8 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:83]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 9 above)
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Changed.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 10]
```diff
@@ -63,11 +66,30 @@ public void testCreateShop() throws Exception {
...
+    public void testGetNewToken_success() {
+        // Arrange
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        UUID shopId = inMemoryVirtualQueueRepository.createShop(shop);
+
+        // Act
+        Token newToken = inMemoryVirtualQueueRepository.getNewToken(shopId);
+        UUID tokenId = newToken.getTokenId();
+        int expectedTokenMapSize = inMemoryVirtualQueueRepository.getTokenMap().size();
+        Integer expectedTokenNumber = inMemoryVirtualQueueRepository.getTokenMap().get(tokenId).getTokenNumber();
```
[the remark this answers is not in the document]
Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:89]: Done.

File: frontendservice/package.json (hunk 4 above)
Comment (KarishmaSinghChauhan) [reply frontendservice/package.json]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 11]
```diff
@@ -63,11 +66,31 @@ public void testCreateShop() throws Exception {
...
     public void testGetWaitingTime() throws Exception {
         long waitingTime = inMemoryVirtualQueueRepository.getWaitingTime();
-        
+
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetNewToken_success() {
+        // Arrange
```
Comment (priytanvani) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:79]: Nit. Add a '.' after every comment.

Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Done.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 12]
```diff
@@ -84,6 +86,26 @@ public void testGetWaitingTime() throws Exception {
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
 
+    @Test
+    public void testGetNewToken_success() throws Exception {
+        // Arrange.
+        CreateShopRequest shop = new CreateShopRequest(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
```
Comment (priytanvani) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:92]: Can we reuse addShopToRepository method for this?

Comment (KarishmaSinghChauhan) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: Done.
